common:
  # 4개의 카메라 토픽을 리스트로 정의
  
  img_topic: "/alphasense/cam3/image_raw" # ONLY_IF num_of_cam == 1 
  img_topics: 
    - "/alphasense/cam3/image_raw" 
    - "/alphasense/cam2/image_raw" 
    - "/alphasense/cam0/image_raw" 
    - "/alphasense/cam4/image_raw" 
  lid_topic: "/hesai/pandar"
  imu_topic: "/alphasense/imu"
  num_of_cam: 4 # 카메라 개수 명시
  img_en: 1
  lidar_en: 1
  ros_driver_bug_fix: false
  sync_time_window: 0.08
  minimum_simultaneous_frame_num: 1
  lidar_window_size: 0.1
  en_sliding_window_ICP: false
  en_cam_backprop: true
  mapping_after_vio: false

extrin_calib:
  # LiDAR to IMU Extrinsics (T_il)
  extrinsic_T: [-0.001, -0.00855, 0.055] 
  extrinsic_R: [0, -1, 0, -1, 0, 0, 0, 0, -1]



  # 앞
  R_cl_2: [ -0.999926,  -0.00670802, 0.0101073,
         -0.0100912, -0.00242564, -0.999946,
          0.00673218,-0.999975, 0.00235777 ]
  P_cl_2: [ -0.0549762, 0.0675401, -0.0520599 ]

  # 위
  R_cl_1: [-0.0042340516721277015, -0.9999897552434941, 0.0016006918802386892,
    0.9999871881804255, -0.004238470700886935, -0.0027674611333545654,
    0.0027742172670246527, 0.0015889537990238962, 0.9999948894591316]
  P_cl_1: [-0.0068618144566339415, -0.007931259020147333, -0.034124943940492]


  R_cl_0: [-0.007396041673023157, 0.9998916894135631, -0.01272430781071447,
    0.002363530935315569, -0.01270714046408309, -0.9999164676625464,
    -0.9999698556902047, -0.007425498159515405, -0.002269292399476633]
  P_cl_0: [-0.0030245656200529533, 0.011180668261220912, -0.05718309342102409]


  R_cl_3: [ 0.000942427662931895, -0.9999880402484476, 0.004799082221662863,
    0.0021900897852221157, -0.004797008865106434, -0.9999860960096802, 
    0.9999971576643774, 0.0009529249803788974,  0.0021855427584387965]
  P_cl_3: [0.006416247252956556, 0.01670064540948574, -0.07488037729385075]

# [Deprecated]

  # extrinsic_R: [1, 0, 0, 0, 1, 0, 0, 0, 1]
  # R_cl_0: [ 0.006708021451800439,  0.9999264200621176,  -0.01010727015365992,
  #          0.002425643641803532,  0.010091197021688258,  0.9999461405473755,
  #           0.9999745590269407, -0.00673217679702115,  -0.0023577731969991467]
  # P_cl_0: [-0.04586422589354697, 0.012631813183337478, -0.05098782892861867]
  # R_cl_1: [ 0.0016556126470597954,  0.9999840642813064,  -0.005397233569448934,
  #           0.0009350089535379302,  0.005395690615039778,  0.9999850060281122,
  #           0.9999981923508761, -0.0016606342845620903,  -0.0009260608798763448]
  # R_cl_2: [ 0.9999897552434941,  0.0042340516721277015, -0.0016006918802386892,
  #           0.004238470700886935,  -0.9999871881804255,  0.0027674611333545654,
  #           -0.0015889537990238962, -0.0027742172670246527,  -0.9999948894591316]
  # R_cl_3: [-0.9998916894135631,  0.007396041673023157,  0.01272430781071447,
  #           0.01270714046408309,  -0.002363530935315569, 0.9999164676625464,
  #           0.007425498159515405,  0.9999698556902047,  0.002269292399476633]
  # R_cl_4: [0.9999880402484476, -0.000942427662931895, -0.004799082221662863,
  #   0.004797008865106434, -0.0021900897852221157,0.9999860960096802, 
  #   -0.0009529249803788974, -0.9999971576643774, -0.0021855427584387965]



time_offset:
  imu_time_offset: 0.0
  img_time_offset: 0.00019
  exposure_time_init: 0.0

preprocess:
  hilti_en: true
  point_filter_num: 1
  filter_size_surf: 0.1 # 0.2
  lidar_type: 5 # HesaiXT32
  scan_line: 32
  blind: 0.6 # 0.1 0.3
 
vio:
  max_iterations: 5
  outlier_threshold: 500
  img_point_cov: 1000
  patch_size: 8
  patch_pyrimid_level: 4
  normal_en: true
  raycast_en: false
  inverse_composition_en: false
  exposure_estimate_en: true
  inv_expo_cov: 0.1

  shiTomasiScore_threshold: 0.0
  min_depth_threshold: 0.0
  max_depth_threshold: 100.0
  min_visual_points: 0
  dismiss_non_outofbound_pixels_from_ref_patch: true
  en_error_LERP_backprop: true

imu:
  imu_en: true
  imu_int_frame: 30
  acc_cov: 0.5 # 0.1
  gyr_cov: 0.01
  b_acc_cov: 0.0001 # 0.1
  b_gyr_cov: 0.0001 # 0.1

lio:
  max_iterations: 5
  dept_err: 0.02
  beam_err: 0.05
  min_eigen_value: 0.0001 # 0.0025
  voxel_size: 0.4
  max_layer: 2
  max_points_num: 100
  layer_init_num: [5, 5, 5, 5, 5]

local_map:  
  map_sliding_en: false
  half_map_size: 100
  sliding_thresh: 8

uav:
  imu_rate_odom: false
  gravity_align_en: false

publish:
  dense_map_en: false
  pub_effect_point_en: false 
  pub_plane_en: false
  pub_scan_num: 1
  blind_rgb_points: 0.0

evo:
  seq_name: "exp09_cupola"
  pose_output_en: true

pcd_save:
  pcd_save_en: false
  colmap_output_en: false # need to set interval = -1
  filter_size_pcd: 0.15
  interval: -1
  # how many LiDAR frames saved in each pcd file;
  # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.
